packages/eas-cli/src/commands/branch/publish.ts

```diff
@@ -473,3 +475,19 @@ function formatUpdateTitle(
     'mmm dd HH:MM'
   )} by ${actorName}, runtimeVersion: ${runtimeVersion}] ${message}`;
 }
+
+async function getEASUpdateURLAsync(exp: ExpoConfig): Promise<string> {
+  const projectId = await getProjectIdAsync(exp);
+  return new URL(projectId, EAS_UPDATE_URL).href;
+}
+
+async function checkEASUpdateURLIsSetAsync(exp: ExpoConfig): Promise<void> {
+  const configuredURL = exp.updates?.url;
+  const expectedURL = await getEASUpdateURLAsync(exp);
+
+  if (configuredURL !== expectedURL) {
+    throw new Error(
+      `The update URL is incorrectly configured for EAS Update. Please set updates.url to ${expectedURL} in your app.json.`
+    );
```

Comment:
One DX question is whether a throw is too aggressive here. I considered just logging a warning, but those are easier to miss and I can't think of a situation in which a user would want to publish without a correctly configured URL.

Reply:
what is the point of that field if we can get that value using other means and overriding it is not allowed?

Reply:
I'm planning to allow overwriting it here: https://github.com/expo/eas-cli/pull/800/files

I didn't want to automatically overwrite it here as the developer maybe wants to use classic updates and the mistake was  note the URL, but using `eas branch:publish` instead of `expo publish`